hs_rest_api2/tests/test_resource_permission.py

```diff
@@ -0,0 +1,179 @@
+        self.assertEqual(response_json['permission'], 'owner')
+
+    def test_get_permission_edit(self):
+        # Test edit permission
+        self.client.logout()
+        self.client.login(username='edituser', password='edituser')
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        response_json = json.loads(response.content.decode())
+        self.assertEqual(response_json['permission'], 'edit')
+
+    def test_get_permission_view(self):
+        # Test view permission
+        self.client.logout()
+        self.client.login(username='viewuser', password='viewuser')
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        response_json = json.loads(response.content.decode())
+        self.assertEqual(response_json['permission'], 'view')
+
+    def test_get_permission_none(self):
+        # Test no permission
+        self.client.logout()
+        self.client.login(username='nopermuser', password='nopermuser')
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        response_json = json.loads(response.content.decode())
+        self.assertEqual(response_json['permission'], 'none')
+
+    def test_get_permission_nonexistent_resource(self):
+        # Test with non-existent resource
+        nonexistent_url = "/hsapi2/resource/abc123/permission/json/"
+        response = self.client.get(nonexistent_url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        response_json = json.loads(response.content.decode())
+        self.assertEqual(response_json['permission'], 'none')
+
+    def test_get_permission_anonymous_user(self):
+        # Test no permission
+        self.client.logout()
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        response_json = json.loads(response.content.decode())
+        self.assertEqual(response_json['permission'], 'none')
+
+    def test_get_permission_group_edit(self):
+        # Test edit permission through group membership
+        self.client.logout()
+        self.client.login(username='groupedituser', password='groupedituser')
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        response_json = json.loads(response.content.decode())
+        self.assertEqual(response_json['permission'], 'edit')
+
+    def test_get_permission_group_view(self):
+        # Test view permission through group membership
+        self.client.logout()
+        self.client.login(username='groupviewuser', password='groupviewuser')
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        response_json = json.loads(response.content.decode())
+        self.assertEqual(response_json['permission'], 'view')
```

Comment:
the way you have written it above, `groupviewuser` should actually have `PrivilegeCodes.CHANGE` not view

Reply:
See my response to your similar comment below.